### PR TITLE
fix: validate header against parent in engine2

### PR DESCRIPTION
**Pull request by Rjected**
Previously we were not validating the header against its parent, and did not do things like timestamp checks.

This causes the hive `engine-api/Invalid NewPayload, Timestamp, Syncing=True, EmptyTxs=False, DynFeeTxs=False` test (and probably more) to pass.

Adds a helper method for getting a sealed header by hash.

### CODE REVIEW

File: crates/engine/tree/src/tree/mod.rs [hunk 1]
```diff
@@ -1396,6 +1417,17 @@ where
...
         };
 
+        // now validate against the parent
+        let parent_block = self.sealed_header_by_hash(block.parent_hash)?.ok_or_else(|| {
+            InsertBlockErrorKindTwo::Provider(ProviderError::HeaderNotFound(
+                block.parent_hash.into(),
+            ))
+        })?;
+        if let Err(e) = self.consensus.validate_header_against_parent(&block, &parent_block) {
+            error!(?block, "Failed to validate header {} against parent: {e}", block.header.hash());
```
Comment (mattsse) [inline crates/engine/tree/src/tree/mod.rs:1427]:
```suggestion
            warn!(?block, "Failed to validate header {} against parent: {e}", block.header.hash());
```

**Review by mattsse — APPROVED** [review]
nice!

File: crates/engine/tree/src/tree/mod.rs [hunk 2]
```diff
@@ -896,6 +896,26 @@ where
...
+            // the hash
+            let block = self
+                .state
+                .tree_state
+                .block_by_hash(hash)
+                // TODO: clone for compatibility. should we return an Arc here?
+                .map(|block| block.as_ref().clone().header);
+
+            Ok(block)
+        }
```
Comment (mattsse) [inline crates/engine/tree/src/tree/mod.rs:916]: should we swap these?

Comment (Rjected) [reply crates/engine/tree/src/tree/mod.rs:916]: yeah, actually, checking memory first would be better

**Review by mattsse — APPROVED** [review]

**Pull request merged**